### PR TITLE
Fixing openmp compile

### PR DESCRIPTION
**Pull request by nikizadehgfdl**
- openmp threads produce non-openmp stats
- openmp 1 thread run has to be a test. Otherwise it would be safer not supporting openmp at all.

### CODE REVIEW

**Comment by nikizadehgfdl** [comment]
This fix is not complete. Dev/gfdl still has openmp issues. I don't know how I was able to compile this branch!!

**Comment by nikizadehgfdl** [comment]
This branch fixes -openmp compilation issues as well as answer changes for threaded runs. 
It closes issue #973 
OMP threaded runs (ocean_nthreads>1) still underperform.

File: src/parameterizations/vertical/MOM_diabatic_aux.F90 [hunk 1]
```diff
@@ -974,9 +974,9 @@ subroutine applyBoundaryFluxesInOut(CS, G, GV, US, dt, fluxes, optics, nsw, h, t
   !$OMP                                  dThickness,dTemp,dSalt,hOld,Ithickness,           &
   !$OMP                                  netMassIn,pres,d_pres,p_lay,dSV_dT_2d,            &
   !$OMP                                  netmassinout_rate,netheat_rate,netsalt_rate,      &
-  !$OMP                                  drhodt,drhods,pen_sw_bnd_rate,SurfPressure,       &
+  !$OMP                                  drhodt,drhods,pen_sw_bnd_rate,                    &
   !$OMP                                  pen_TKE_2d,Temp_in,Salin_in,RivermixConst)        &
-  !$OMP                     firstprivate(start,npts)
+  !$OMP                     firstprivate(start,npts,SurfPressure)
```
Comment (nikizadehgfdl) [inline src/parameterizations/vertical/MOM_diabatic_aux.F90]: This fixes the answers mismatch between openmp and non-openmp execs.

**Comment by marshallward** [comment]
I just tested an OpenMP build and am also seeing errors in the `MOM_geothermal.F90` code:
```
ftn -Duse_libMPI -Duse_netCDF -DSPMD -Duse_LARGEFILE  -I/opt/cray/pe/netcdf/4.4.0/GNU/5.1/include -fcray-pointer -fdefault-real-8 -fdefault-double-8 -Waliasing -ffree-line-length-none -fno-range-check -O2 -fno-expensive-optimizations -fopenmp -I /lustre/f2/dev/gfdl/Marshall.Ward/dev/mom6/.testing//../deps/fms/build  -c -I/lustre/f2/dev/gfdl/Marshall.Ward/dev/mom6/.testing//../config_src/dynamic_symmetric -I/lustre/f2/dev/gfdl/Marshall.Ward/dev/mom6/.testing//../src/framework	/lustre/f2/dev/gfdl/Marshall.Ward/dev/mom6/.testing//../src/parameterizations/vertical/MOM_geothermal.F90
/lustre/f2/dev/gfdl/Marshall.Ward/dev/mom6/.testing//../src/parameterizations/vertical/MOM_geothermal.F90:158:60:

                   .or. CS%id_internal_heat_temp_tendency > 0
                                                            1
Error: Unexpected assignment statement at (1)
/lustre/f2/dev/gfdl/Marshall.Ward/dev/mom6/.testing//../src/parameterizations/vertical/MOM_geothermal.F90:161:60:

                   .or. CS%id_internal_heat_temp_tendency > 0
                                                            1
Error: Unexpected assignment statement at (1)
/lustre/f2/dev/gfdl/Marshall.Ward/dev/mom6/.testing//../src/parameterizations/vertical/MOM_geothermal.F90:163:65:

   if (CS%id_internal_heat_heat_tendency > 0) work_3d(:,:,:) = 0.0
                                                                 1
Error: Unexpected simple IF statement at (1)
/lustre/f2/dev/gfdl/Marshall.Ward/dev/mom6/.testing//../src/parameterizations/vertical/MOM_geothermal.F90:164:39:

   if (compute_h_old) h_old(:,:,:) = 0.0
                                       1
Error: Unexpected simple IF statement at (1)
/lustre/f2/dev/gfdl/Marshall.Ward/dev/mom6/.testing//../src/parameterizations/vertical/MOM_geothermal.F90:165:39:

   if (compute_T_old) T_old(:,:,:) = 0.0
                                       1
Error: Unexpected simple IF statement at (1)
```

Assuming it's not too much trouble, I think we should probably add this to the test suite, if only to check that it builds.

**Comment by marshallward** [comment]
Gaea regression tests: https://gitlab.gfdl.noaa.gov/ogrp/MOM6/pipelines/9074

**Comment by marshallward** [comment]
Gaea regressions have passed.

I can also build this internally although I am seeing some differences in answers for some of the TCs in the new test suite.  But I will look into those in a different PR.

**Pull request merged**